_posts/2015-08-24-Cocoa-Architecture:-Hybrid-Apps.markdown

```diff
@@ -0,0 +1,68 @@
+
+----
+
+### What is a Hybrid App
+
+I gave the widest possible definition above, so let's dig in as this can be a contentious. There is a whole spectrum of which an app can be classed as a hybrid app. This ranges from; more or less web-tech everywhere to 100% Objective-C.
+
+A great example of the furthest to the web-side is [ATOM](https://atom.io), the text editor. It's on the extreme side because all of the user interface is built using HTML + CSS, and almost all of the app is in javascript. The trade-off for them is that they can easy write cross-platform code, that will work reliably with technology that the vast majority of programmers use.
+
+An example of a purely native application would be [Energy](https://github.com/artsy/energy/). It's over a hundred thousand plus lines of Objective-C. Yet under the hood, there's definitely some web-tech there. Prior to iOS7 `UILabel`s, `UITextField`s and `UITextField`s [used WebKit for rendering](http://www.objc.io/issues/5-ios7/getting-to-know-textkit/). Not to mention that when a Partner send's an email via Energy, the editor is a `UIWebView`. However, from the app developer's perspective they are creating native interactions that are consistent with the rest of the operating system's behavior.
+
+### Eigen
+
+When we started building Eigen, it was very obvious that we had a Sisyphean task ahead of us. We wanted to take the core ideas of the Artsy website, _The Art World Online_, and convert it into Mobile, _The Art World in Your Pocket_.
+
+That's not impossible, but the mobile team was a fraction of the Artsy development team. Any new features added to the website would need a mobile equivalent, and given the speed in which web developer's can ship, we'd need to outnumber them to stand a chance at keeping up.
+
+So, we opted for building a highly integrated mobile website at the same time, it would use the same design language as the iOS app and can act as a way for Android devices to access Artsy.
+
+### Techniques for Doing It Well
+
+So we'll be talking about our [`ARInteralMobileViewContrller`](https://github.com/artsy/eigen/blob/6bb44a01c1b23fb8e92c645c3091fd33725743c3/Artsy/View_Controllers/Web_Browsing/ARInternalMobileWebViewController.m) - which currently relies on `UIWebView` but is in [the process of](https://github.com/artsy/eigen/pull/606) migrating to `WKWebkit`.
+
+* Scroll like an [iOS app](https://github.com/artsy/eigen/blob/6bb44a01c1b23fb8e92c645c3091fd33725743c3/Artsy/View_Controllers/Web_Browsing/ARExternalWebBrowserViewController.m#L39) by setting the web view's `scrollView.decelerationRate = UIScrollViewDecelerationRateNormal`.
+
+* Use a simple design language to avoid the [uncanny valley](http://tvtropes.org/pmwiki/pmwiki.php/Main/UncannyValley). Care about using the same typographical rules on everything including tabs, buttons and switches.
```

Comment:
"same typographical rules" could be a link to one our pods, like `Artsy+UILabels`.